backend/api/v2beta1/recurringrun.proto

```diff
@@ -0,0 +1,318 @@
+// Copyright 2018 The Kubeflow Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+syntax = "proto3";
+
+option go_package = "github.com/kubeflow/pipelines/backend/api/go_client";
+package v2beta1;
+
+import "google/api/annotations.proto";
+import "google/protobuf/timestamp.proto";
+import "google/protobuf/empty.proto";
+import "protoc-gen-swagger/options/annotations.proto";
+import "backend/api/v2beta1/error.proto";
+
+option (grpc.gateway.protoc_gen_swagger.options.openapiv2_swagger) = {
+  responses: {
+    key: "default";
+    value: {
+      schema: {
+        json_schema: {
+          ref: ".api.Status";
+        }
+      }
+    }
+  }
+  // Use bearer token for authorizing access to job service.
+  // Kubernetes client library(https://kubernetes.io/docs/reference/using-api/client-libraries/)
+  // uses bearer token as default for authorization. The section below
+  // ensures security definition object is generated in the swagger definition.
+  // For more details see https://github.com/OAI/OpenAPI-Specification/blob/3.0.0/versions/2.0.md#securityDefinitionsObject
+  security_definitions: {
+    security: {
+      key: "Bearer";
+      value: {
+        type: TYPE_API_KEY;
+        in: IN_HEADER;
+        name: "authorization";
+      }
+    }
+  }
+  security: {
+    security_requirement: {
+      key: "Bearer";
+      value: {};
+    }
+  }
+};
+
+
+service RecurringRunService {
+  // Creates a new recurring run in an experiment, given the experiment ID.
+  rpc CreateRecurringRunByExperiment(CreateRecurringRunByExperimentRequest) returns (RecurringRun) {
+    option (google.api.http) = {
+      post: "/apis/v2beta1/experiments/{id}/recurringruns"
+      body: "recurringrun"
+    };
+  }
+
+  // Creates a new recurring run in an experiment, given the experiment ID.
+  rpc CreateRecurringRunByExperimentName(CreateRecurringRunByExperimentNameRequest) returns (RecurringRun) {
+    option (google.api.http) = {
+      post: "/apis/v2beta1/experiments/names/{name}/recurringruns"
+      body: "recurringrun"
+    };
+  }
+
+  // Finds a specific recurring run by ID.
+  rpc GetRecurringRun(GetRecurringRunRequest) returns (RecurringRun) {
+    option (google.api.http) = {
+      get: "/apis/v2beta1/recurringruns/{id}"
```

Comment:
Isn't recurring run a sub-resource under experiments? Experiments should always be in the path then.